### ISSUE TITLE
[Tune] Ray.Tune is not finding saved checkpoints

### ISSUE THREAD

**Issue by ericbhanson** [ray-project/ray#36062]
### What happened + What you expected to happen

Trying to adapt the example at https://pytorch.org/tutorials//beginner/hyperparameter_tuning_tutorial.html to restore experiments in case of a machine failure, as I'm using free cloud resources that can be interrupted before my model is finished to run my code. When running the experiment for the first time, I can see that Ray is storing checkpoint information in my specified storage directory, but when I restart the notebook kernel and run the script again, Ray says `No local checkpoint was found. Ray Tune will now start a new experiment.`

Attached are two of the JSON files that Ray generated during its first run.

[JSON files.zip](https://github.com/ray-project/ray/files/11646683/JSON.files.zip)

### Versions / Dependencies

Ray: 2.4.0
Python: 3.9.16

### Reproduction script

result = tune.run(
        partial(train_cifar, data_dir=data_dir),
        resources_per_trial={"cpu": 5, "gpu": gpus_per_trial},
        config=config,
        num_samples=num_samples,
        scheduler=scheduler,
        resume='AUTO',
        local_dir=checkpoints_dir
    )

### Issue Severity

High: It blocks me from completing my task.

**Comment by justinvyu** [comment]
@ericbhanson

You'll want to set `name` in `tune.run` in order for Tune to know to look for the experiment at a fixed path.

```python
# The experiment results will be stored at `~/ray_results/experiment_dirname`
tune.run(..., name="experiment_dirname", local_dir="~/ray_results", resume="AUTO")
```

Let me know if that solves your problem.
